mhs-reference-implementation/mhs/SDSLookup/tests/test_sdsclient.py

```diff
@@ -0,0 +1,68 @@
+from unittest import TestCase
+import mhs.SDSLookup.sds as sds
+import mhs.SDSLookup.tests.ldap_mocks as mocks
+from utilities.test_utilities import async_test
+
+NHS_SERVICES_BASE = "ou=services, o=nhs"
+
+MHS_OBJECT_CLASS = "nhsMhs"
+
+PARTY_KEY = "A91461-9199084"
+SERVICE_ID = "urn:nhs:names:services:psis:MCCI_IN010000UK13"
+ODS_CODE = "A91461"
+
+
+class TestSDSClient(TestCase):
+
+    @async_test
+    async def test_mhs_details_lookup(self):
+        server, connection = mocks.fake_ldap_server()
+
+        client = sds.SDSClient()
+        client.server = server
+        client.connection = connection
+
+        result = await client._mhs_details_lookup(PARTY_KEY, SERVICE_ID)
+
+        self.assertIsNotNone(result)
+        self.assertEqual(result[0]['attributes']['nhsMHSEndPoint'], ['https://vpn-client-1411.opentest.hscic.gov.uk/'])
+
+    @async_test
+    async def test_accredited(self):
+        server, connection = mocks.fake_ldap_server()
+        client = sds.SDSClient()
+        client.server = server
+        client.connection = connection
+
+        result = await client._accredited_system_lookup(ODS_CODE, SERVICE_ID)
+        assert result
+        assert result[0]['attributes']['nhsMHSPartyKey'] == 'A91461-9199084'
```

Comment:
Again, we want to verify that no other attributes have been returned.

Reply:
Check that exactly 1 attribute is returned (party key)